tools/istio-iptables/pkg/builder/rules_builder.go

```diff
@@ -0,0 +1,52 @@
+package builder
+
+type Rule struct {
+	chain  *string
+	table  *string
+	params []string
+}
+
+type Rules struct {
+	rules   []*Rule
+	rulesv6 []*Rule
+}
+
+type Builder interface {
+	AddRule(chain *string, table *string, params ...string) Builder
+	AddRuleV6(chain *string, table *string, params ...string) Builder
+	Build() []*string
+	BuildV6() []*string
+	BuildIptablesRestoreFormat() *string
+	BuildIptablesV6RestoreFormat() *string
```

Comment:
- To make the names symmetric, I'd include `V4` in the names of non-`V6` methods, except for the `Build...` methods where I'd stick to the `iptables-restore` and `ip6tables-restore` command names.
- No need to use pointers to `string`, it's already a reference type, so `*string` -> `string` everywhere.
- The `Build...` methods are not building a "format", so the `Build...Format` seems weird.

```suggestion
	AddV4Rule(chain string, table string, params ...string) IptablesBuilder
	AddV6Rule(chain string, table string, params ...string) IptablesBuilder
	BuildV4Rules() []string
	BuildV6Rules() []string
	BuildIptablesRestore() string
	BuildIp6tablesRestore() string
```

Reply:
fixed